### PR TITLE
ui: items facets and sorting

### LINKED ISSUE #68: Back-office ui: integrate search aggregations and sorting in items
Integrate aggregation and sorting in items search page.

- [ ] setup sorting and aggregations in config.py
- [ ] inject configuration in react app
- [ ] integrate React-SearchKit sorting and aggregations

### PR DESCRIPTION
**Pull request by ntarocco**
* uses react-searchkit 0.3.0
* inject invenio configuration in React app
* adds sorting and facets in backoffice items search
* upgrades npm dependencies for security fixes
* fixes dependencies
* closes #68

### CODE REVIEW

**Comment by ntarocco** [comment]
PR is passing tests, but Travis got stuck with a wrong git repo reference... please ignore it.

File: invenio_app_ils/ui/backoffice/src/pages/ItemsSearch/ItemsSearch.js [hunk 1]
```diff
@@ -44,6 +47,24 @@ const resultsPerPageValues = [
   },
 ];
 
+const config = invenioConfig.items.search;
```
Comment (zzacharo) [inline invenio_app_ils/ui/backoffice/src/pages/ItemsSearch/ItemsSearch.js]: Split this part to a `config.js` file?

Comment (zzacharo) [reply invenio_app_ils/ui/backoffice/src/pages/ItemsSearch/ItemsSearch.js]: I would say to expose them under a const config variable, so here we can reference them as `config.xxx`, just to know immediately from where they are coming.

File: invenio_app_ils/ui/backoffice/src/pages/ItemsSearch/ItemsSearch.js [hunk 2]
```diff
@@ -100,46 +121,57 @@ export class ItemsSearch extends Component {
...
-          compact
-          options={values}
-          value={currentSize}
-          onChange={(e, { value }) => onChange(value)}
+  _renderResultsSorting = () => {
+    return sortByValues.length ? (
+      <div className="n-per-page-sorting">
+        <span className="before">Show</span>
+        <ResultsPerPage
+          values={resultsPerPageValues}
```
Comment (zzacharo) [inline invenio_app_ils/ui/backoffice/src/pages/ItemsSearch/ItemsSearch.js]: Can you capitalize that so we can identify that is a global variable and not passed as a prop?

File: invenio_app_ils/ui/backoffice/src/pages/ItemsSearch/ItemsSearch.scss [hunk 3]
```diff
@@ -3,8 +3,13 @@
 }
 
 .items-search-container {
-  .results-per-page {
-    padding: 0 0.5em;
+  .n-per-page-sorting {
```
Comment (zzacharo) [inline invenio_app_ils/ui/backoffice/src/pages/ItemsSearch/ItemsSearch.scss]: Omit n? I would wait to be a function that you pass the n value and return a dynamic class...

File: invenio_app_ils/ui/backoffice/src/pages/ItemsSearch/ItemsSearch.js (hunk 1 above)
Comment (ntarocco) [reply invenio_app_ils/ui/backoffice/src/pages/ItemsSearch/ItemsSearch.js]: Done!

File: invenio_app_ils/ui/backoffice/src/pages/ItemsSearch/ItemsSearch.js (hunk 2 above)
Comment (ntarocco) [reply invenio_app_ils/ui/backoffice/src/pages/ItemsSearch/ItemsSearch.js]: Done!

File: invenio_app_ils/ui/backoffice/src/pages/ItemsSearch/ItemsSearch.scss (hunk 3 above)
Comment (ntarocco) [reply invenio_app_ils/ui/backoffice/src/pages/ItemsSearch/ItemsSearch.scss]: Done!

**Review by zzacharo — APPROVED** [review]

**Pull request merged**